source/object-entries.ts

```diff
@@ -0,0 +1,21 @@
+/**
+A strongly-typed version of `Object.entries()`.
+
+This is useful since `Object.entries()` always returns an array of Array<[string, T]>. This function returns a strongly-typed array of the entries of the given object.
+
+- [TypeScript issues about this](https://github.com/microsoft/TypeScript/pull/12253)
+
+@example
+```
+import {objectEntries} from 'ts-extras';
+
+const stronglyTypedEntries = objectEntries({a: 1, b: 2, c: 3}); // =>  Array<["a" | "b" | "c", number]>
+const untypedEntries = Object.entries(items); // => Array<[string, number]>
+```
+*/
+export function objectEntries<Type extends Record<string, unknown>, Key extends Extract<keyof Type, string>>(
```

Comment:
```suggestion
export function objectEntries<Type extends Record<never, never>>(value: Type): Array<[keyof Type, Type[keyof Type]]> {
```
Not sure if it's possible but you could make this even stricter by using [mapped types](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html).

Reply:
I wonder if there's a reason why it should be `Record<never, never>` instead of `Record<string, unknown>`?

`objectKeys()` is using `Record<string, unknown>` https://github.com/sindresorhus/ts-extras/blob/main/source/object-keys.ts

My suggestion.

```ts
export function objectEntries<Type extends Record<string, unknown>>(value: Type): Array<[keyof Type, Type[keyof Type]]>
```

Reply:
Now that I think about it, it might be better to use `Record<never, V>` but have the return object be
```ts
Record<`${Extract<keyof Type, string | number>}`, V>
```
since symbol types are excluded from `Object.keys/entries`.

Reply:
If it is to exclude the symbol type, how about the following?

The input type parameter is open for all Record types, but returns except for the symbol key and the value type of the symbol key.

```ts
 export function objectEntries<Type extends Record<PropertyKey, unknown>, Key extends Exclude<keyof Type, symbol>>(value: Type): Array<[Key, Type[Key]]> {
   return Object.entries(value) as Array<[Key, Type[Key]]>;
 }

objectEntries({ 
  1: 123, 
  'hello': 'bye', 
  [Symbol('symbol')]: true
}); // ["hello" | 1, string | number][]
```

Reply:
I have committed the code above. Could you review it and resolve the conversation?

Reply:
I'm not able to resolve this conversation but the changes look good!